pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -924,3 +924,88 @@ func removeNodeIPAddressesFromBackendPool(
 
 	return changed
 }
+
+type backendPoolTypePodIP struct {
+	*Cloud
+}
+
+func newBackendPoolTypePodIP(c *Cloud) BackendPool {
+	return &backendPoolTypePodIP{c}
+}
+
+func (bpi *backendPoolTypePodIP) CleanupVMSetFromBackendPoolByCondition(_ context.Context, _ *armnetwork.LoadBalancer, _ *v1.Service, _ []*v1.Node, _ string, _ func(string) bool) (*armnetwork.LoadBalancer, error) {
+	return nil, nil
+}
+
+func (bpi *backendPoolTypePodIP) EnsureHostsInPool(_ context.Context, _ *v1.Service, _ []*v1.Node, _ string, _ string, _ string, _ string, _ *armnetwork.BackendAddressPool) error {
+	return nil
+}
+
+func (bpi *backendPoolTypePodIP) GetBackendPrivateIPs(_ context.Context, _ string, service *v1.Service, lb *armnetwork.LoadBalancer) ([]string, []string) {
+	serviceName := getServiceName(service)
+
+	lbBackendPoolName, err := bpi.getBackendPoolNameForCLBService(service)
+	if err != nil {
+		klog.Errorf("bpi.GetBackendPrivateIPs: failed to get backend pool name for service %q, error: %s", service.Name, err.Error())
+		return nil, nil
+	}
+
+	if lb.Properties == nil || lb.Properties.BackendAddressPools == nil {
+		return nil, nil
+	}
+
+	backendPrivateIPs := utilsets.NewString()
+	for _, bp := range lb.Properties.BackendAddressPools {
```

Comment:
It looks similiar with nodeIP backend pool GetBackendPrivateIPs. Is is possible to reuse some logic as a separate func?